### DISCUSSION TITLE
怎样挂载SharePoint站点呢？

### DISCUSSION THREAD

**Discussion by LordHumphrey** [spencerwooo/onedrive-vercel-index#339] [Q&A]
翻了文档好像没有翻到，类似于这种 

```
https://*.sharepoint.com/sites/*
```

**Comment by spencerwooo** [comment]
没试过，但是可能得改 `api.config.js` 里面的 `authApi` 和 `driveApi`？

https://onedrive-vercel-index.spencerwoo.com/docs/getting-started#modify-api-settings-optional

**Reply by spencerwooo** [reply]
貌似 SharePoint 的话权限请求的也不太一样？好像要 `sites.read.all` 而不是 `files.read.all`？

**Reply by LordHumphrey** [reply]
改了，没有用的，所以能不能加个多盘切换的功能，或者能不能把SharePoint做成一个目录，点进去就是网站文件内容呢，可以考虑一下这个功能，谢谢啦~

**Reply by spencerwooo** [reply]
E5 有 SharePoint 用吗？我最近才搞了 E5，不太清楚，之前写这个项目都是用的 OneDrive 个人版（还是国际版），各个版本的测试账号至少得有才能谈支持别的东西了

**Reply by LordHumphrey** [reply]
没有呀，我是A1，但是如果用Sharepoint的话，可以通过添加快捷方式到OneDrive当中，这样只用一个账号就能查看，搜索所有文件了，不用来回切换账号了，本地配合Everything还是很舒服的，如果再加一个index更美滋滋啦

**Reply by opoet** [reply]
> E5 有 SharePoint 用吗？我最近才搞了 E5，不太清楚，之前写这个项目都是用的 OneDrive 个人版（还是国际版），各个版本的测试账号至少得有才能谈支持别的东西了

E5貌似是有的，应该是在这里创建共享库，我这个就是E5

![image](https://user-images.githubusercontent.com/30021379/155285224-44aa6062-d579-4019-af21-9704851891aa.png)

**Comment by Ovler-Young** [comment]
可以参考[旧项目文档](https://github.com/spencerwooo/onedrive-cf-index/blob/main/README-CN.md#%E7%94%9F%E6%88%90-onedrive-api-%E4%BB%A4%E7%89%8C) 中方法，通过 [ms-graph-cli](https://github.com/beetcb/ms-graph-cli) 工具生成 driveapi ，在这个项目也可以使用；此外，权限似乎是一样的。

**Reply by iskycc** [reply]
你知道那个userPrincipalName填的啥嘛？

**Reply by Ovler-Young** [reply]
> 你知道那个userPrincipalName填的啥嘛？

当时我把检测的代码先删了搞过去的…

**Reply by Ovler-Young** [reply]
> > 你知道那个userPrincipalName填的啥嘛？
> 
> 当时我把检测的代码先删了搞过去的…

我记得当时 api 获取好像是undefined

**Reply by iskycc** [reply]
> > > 你知道那个userPrincipalName填的啥嘛？
> > 
> > 
> > 当时我把检测的代码先删了搞过去的…
> 
> 我记得当时 api 获取好像是undefined

好的，昨天作者已经告诉我怎么改检测代码了，发送的请求确实没有返回userPrincipalName

**Comment by iskycc** [comment]
你可以使用这个工具 https://github.com/beetcb/ms-graph-cli 以及参考一下这个讨论 https://github.com/spencerwooo/onedrive-vercel-index/discussions/479 就可以部署成功啦

**Reply by myl7** [reply]
在这个讨论中 name 这个字段是可以用户自己修改的，也就是说攻击者可以设一个相同的 name 来越过限制
不过相应的 sharepoint site 支持已经有在讨论了，可以稍等一下

**Reply by 212-mei** [reply]
> 在这个讨论中 name 这个字段是可以用户自己修改的，也就是说攻击者可以设一个相同的 name 来越过限制 不过相应的 sharepoint site 支持已经有在讨论了，可以稍等一下

没事嘛，name字段可以自己修改，可以用其他字段嘛，可以部署成功而且没有什么问题就好了，sharepoint受众比较小，如果愿意折腾的自己改一下就好了，你们再特地去支持这个，不如花时间先完善和修复一下其他的问题？

**Reply by myl7** [reply]
其他哪些问题呢？现在放在 issue 里的应该都是没法马上处理的。

**Reply by 212-mei** [reply]
> 其他哪些问题呢？现在放在 issue 里的应该都是没法马上处理的。

嗯，比如说好几个讨论都表明虽然永久直链增强了可读性，但是出现了无法下载或者无法使用的情况？

**Reply by myl7** [reply]
最新的 commit 已经修复了

**Reply by iskycc** [reply]
那如果你有时间添加支持SharePoint站点的话，应该也可以直接适用于世纪互联吧？

**Reply by 212-mei** [reply]
> 最新的 commit 已经修复了

好的，现在重新部署就好了吗？

**Reply by myl7** [reply]
git merge 到最新的 commit 以后就行了，repo 页面上 fetch upstream 也行
